### ISSUE TITLE
Identify requirements to migrate flutter/flutter from master to main

### ISSUE THREAD

**Issue by godofredoc** [flutter/flutter#121564]
This is a non comprehensive list of what is required to deprecate master branch:

- [ ] Mirror master branch to main
- [ ] validate full support for main alias in flutter tool
- [ ] validate full support for main alias in analytics
- [ ] update frob tooling to use main
- [ ] update conductor to use main
- [ ] update master references across flutter, engine and website
- [ ] create main-api-docs gcp project and update all the workflows to publish docs there
- [ ] update master references in api docs and/or samples

**Comment by godofredoc** [comment]
@christopherfujino can you please help us confirm if there are more changes required in the tool to fully support main as a channel?

**Comment by godofredoc** [comment]
\cc @mit-mit @timsneath @Hixie

**Comment by christopherfujino** [comment]
> @christopherfujino can you please help us confirm if there are more changes required in the tool to fully support main as a channel?

added "migrate existing SDK users on the `master` channel/branch to `main`"

**Comment by godofredoc** [comment]
@jayoung-lee that can you please help us confirm the "main" alias is in place for the flutter analytics?

**Comment by godofredoc** [comment]
@ricardoamador @drewroengoogle can you please help with `create main-docs-flutter-dev gcp project and update all the workflows to publish docs there master-docs-flutter-dev`

**Comment by godofredoc** [comment]
@CaseyHillers are there any references in frob that need to be updated to main?

**Comment by CaseyHillers** [comment]
Google Testing (FRoB) has a few references, but they're non-blocking. It queries commits on master, which as long as we leave the mirror up, will be fine.

**Comment by ricardoamador** [comment]
Will do.

**Comment by jayoung-lee** [comment]
The analytics will record whatever channel name it sees, but the dashboard is already showing `master` as `main`. Would that be sufficient?

**Comment by godofredoc** [comment]
> The analytics will record whatever channel name it sees, but the dashboard is already showing `master` as `main`. Would that be sufficient?

Yes, accounting for main channel in analytics should be enough, thanks!

**Comment by godofredoc** [comment]
> ey're non-blocking. It queries commits on master, which as long as we leave the mirror up, will be fine.

For now the mirror will be kept but the end goal is to completely remove the master branch.

**Comment by timsneath** [comment]
It's worth noting that `main` is not fully working as a mirror of `master` yet, for anyone who might be tempted to try this early.

```
$ flutter channel
Flutter channels:
* master
  main
  beta
  stable
$ flutter channel main
Switching to flutter channel 'main'...
git: fatal: 'origin/main' is not a commit and a branch 'main' cannot be created from it
Switching channels failed with error code 128.
$ flutter channel master
Switching to flutter channel 'master'...
git: Already on 'master'
git: Your branch is up to date with 'upstream/master'.
Successfully switched to flutter channel 'master'.
To ensure that you're on the latest build from this channel, run 'flutter upgrade'
```

**Comment by godofredoc** [comment]
> It's worth noting that `main` is not fully working as a mirror of `master` yet, for anyone who might be tempted to try this early.
> 
> ```
> $ flutter channel
> Flutter channels:
> * master
>   main
>   beta
>   stable
> $ flutter channel main
> Switching to flutter channel 'main'...
> git: fatal: 'origin/main' is not a commit and a branch 'main' cannot be created from it
> Switching channels failed with error code 128.
> $ flutter channel master
> Switching to flutter channel 'master'...
> git: Already on 'master'
> git: Your branch is up to date with 'upstream/master'.
> Successfully switched to flutter channel 'master'.
> To ensure that you're on the latest build from this channel, run 'flutter upgrade'
> ```

This is strange, is your original checkout from more than 10 months ago? the main branch has been mirroring master from ~1 year.

Does a `git checkout master && git fetch upstream && git merge upstream/master` fixes the problem?

**Comment by christopherfujino** [comment]
> git: fatal: 'origin/main' is not a commit and a branch 'main' cannot be created from it

I think this is actually a problem with your setup, it looks like the default fetch remote is origin (which I'm guessing is your fork), while the official channel "master" is set up to fetch from upstream (which I'm guessing is flutter/flutter).

**Comment by christopherfujino** [comment]
Actually, the problem is we hard-code the remote name to be "origin": https://github.com/flutter/flutter/blob/main/packages/flutter_tools/lib/src/commands/channel.dart#L174

However, for non-contributors, the case should be that "origin" will be flutter/flutter.

**Comment by christopherfujino** [comment]
> Is this "you're a really weird and eccentric outlier, and we can ignore your specific machine configuration", or is this "you did `git clone` over a year ago", in which case this is a scenario we need to manage?

The former. @timsneath Since I know you like to reproduce the average developer workflow, one thing you could do to get the out of the box flutter SDK experience is:

```
$ git clone git@github.com:flutter/flutter.git /path/to/flutter # clone the upstream remote as "origin"
$ cd /path/to/flutter
$ git checkout stable && git checkout beta # ensure the "stable" and "beta" branches are present locally, tracking "origin"
$ git remote add downstream git@github.com:timsneath/flutter # name your fork downstream
```

Now, you can open PRs by pushing your local branches to the `downstream` remote, but the official release channels will fetch from "origin", and any situations where the git tool implicitly assumes "origin" will work as they do for users who download the prebuilt archive from the website.

**Comment by timsneath** [comment]
Ah, OK. 

Yes, I see that I cloned my fork and set an upstream, as you surmised:
```
$ git remote -v
origin	git@github.com:timsneath/flutter.git (fetch)
origin	git@github.com:timsneath/flutter.git (push)
upstream	git@github.com:flutter/flutter.git (fetch)
upstream	git@github.com:flutter/flutter.git (push)
```

But that's what the wiki suggests, rather than your guide above: https://github.com/flutter/flutter/wiki/Setting-up-the-Framework-development-environment#set-up-your-environment

Should we change the wiki?

**Comment by sgehrman** [comment]
shocked that people really think this is necessary. Will you change it again when someone finds a reason to be offended by "main"?

**Comment by Hixie** [comment]
Regardless of the quite valid reasons to want our project to be welcoming to all reasonable people, including and especially those who have been disenfranchised and oppressed, there are pragmatic reasons to want to maintain consistency across all our repositories and consistency with the default branch name across GitHub.
